backend/baseclass.pm

```diff
@@ -900,6 +901,12 @@ sub set_tags_to_assert {
     return {tags => \@tags};
 }
 
+sub set_assert_screen_timeout {
+    my ($self, $timeout) = @_;
+    return bmwqemu::diag('set_assert_screen_timeout called with non-numeric timeout') unless looks_like_number($timeout);
```

Comment:
shouldn't we die here?

Reply:
*bump*

Reply:
Yes. Or return 0/1 to distinguish wrong/successful calls. I'll add this to my TODOs for refactoring and general improvements.